### PR TITLE
[office-js][office-js-preview] (Outlook) Add saveAsync behavior of shared mailboxes in Outlook on the web

### PR DESCRIPTION
**Pull request by samantharamon**
- [X] Use a meaningful title for the pull request. Include the name of the package modified.
- [ ] Test the change in your own code. (Compile and run.)
- [ ] [Add or edit tests](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#my-package-teststs) to reflect the change.
- [X] Follow the advice from the [readme](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#make-a-pull-request).
- [X] Avoid [common mistakes](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#common-mistakes).
- [X] [Run `npm test <package to test>`](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#running-tests).

If changing an existing definition:
- [ ] Provide a URL to documentation or source code which provides context for the suggested changes: None.
- [ ] If this PR brings the type definitions up to date with a new version of the JS library, update the version number in the header.

### CODE REVIEW

**Review by Rick-Kirkham — APPROVED** [review]

File: types/office-js-preview/index.d.ts [hunk 1]
```diff
@@ -10914,76 +10914,77 @@ declare namespace Office {
         /**
          * Asynchronously saves an item.
          *
-         * When invoked, this method saves the current message as a draft and returns the item ID via the callback function.
-         * In Outlook on the web or Outlook in online mode, the item is saved to the server.
-         * In Outlook in cached mode, the item is saved to the local cache.
+         * Since appointments have no draft state, if `saveAsync` is called on an appointment in compose mode, the item is saved as a normal
+         * appointment on the user's calendar. For new appointments that have not been saved before, no invitation is sent.
```
Comment (ElizabethSamuel-MSFT) [inline types/office-js-preview/index.d.ts:10918]:
```suggestion
         * appointment on the user's calendar. For new appointments that haven't been saved before, no invitation is sent.
```

File: types/office-js-preview/index.d.ts [hunk 2]
```diff
@@ -10914,76 +10914,77 @@ declare namespace Office {
...
          * **Errors**:
          *
          * - `InvalidAttachmentId`: The attachment identifier does not exist.
          *
          * @param options - An object literal that contains one or more of the following properties:-
          *        `asyncContext`: Developers can provide any object they wish to access in the callback function.
-         * @param callback - When the method completes, the function passed in the `callback` parameter is called with a single parameter of
-         *                   type `Office.AsyncResult`.
+         * @param callback - When the method completes, the function passed in the `callback` parameter is called with a single parameter, `asyncResult`,
+         *                   which is an `Office.AsyncResult` object. The ID of the appointment is returned in the `asyncResult.value` property.
```
Comment (ElizabethSamuel-MSFT) [inline types/office-js-preview/index.d.ts:10950]:
```suggestion
         *                   which is an `Office.AsyncResult` object. The appointment ID is returned in the `asyncResult.value` property.
```

File: types/office-js-preview/index.d.ts [hunk 3]
```diff
@@ -10914,76 +10914,77 @@ declare namespace Office {
...
          */
         saveAsync(options: Office.AsyncContextOptions, callback: (asyncResult: Office.AsyncResult<string>) => void): void;
         /**
          * Asynchronously saves an item.
          *
-         * When invoked, this method saves the current message as a draft and returns the item ID via the callback function.
-         * In Outlook on the web or Outlook in online mode, the item is saved to the server. In Outlook in cached mode, the item is saved to the local cache.
+         * Since appointments have no draft state, if `saveAsync` is called on an appointment in compose mode, the item is saved as a normal
+         * appointment on the user's calendar. For new appointments that have not been saved before, no invitation is sent.
```
Comment (ElizabethSamuel-MSFT) [inline types/office-js-preview/index.d.ts:10957]:
```suggestion
         * appointment on the user's calendar. For new appointments that haven't been saved before, no invitation is sent.
```

File: types/office-js-preview/index.d.ts [hunk 4]
```diff
@@ -10914,76 +10914,77 @@ declare namespace Office {
...
-         * **{@link https://learn.microsoft.com/office/dev/add-ins/outlook/outlook-add-ins-overview#extension-points | Applicable Outlook mode}**: Appointment Organizer
-         *
          * **Errors**:
          *
          * - `InvalidAttachmentId`: The attachment identifier does not exist.
          *
-         * @param callback - When the method completes, the function passed in the callback parameter is called with a single parameter of
-         *                   type `Office.AsyncResult`.
+         * @param callback - When the method completes, the function passed in the `callback` parameter is called with a single parameter, `asyncResult`,
+         *                   which is an `Office.AsyncResult` object. The ID of the appointment is returned in the `asyncResult.value` property.
```
Comment (ElizabethSamuel-MSFT) [inline types/office-js-preview/index.d.ts:10987]:
```suggestion
         *                   which is an `Office.AsyncResult` object. The appointment ID is returned in the `asyncResult.value` property.
```

File: types/office-js-preview/index.d.ts [hunk 5]
```diff
@@ -16676,33 +16662,35 @@ declare namespace Office {
...
          * **Errors**:
          *
          * - `InvalidAttachmentId`: The attachment identifier does not exist.
          *
          * @param options - An object literal that contains one or more of the following properties:-
          *        `asyncContext`: Developers can provide any object they wish to access in the callback function.
-         * @param callback - When the method completes, the function passed in the `callback` parameter is called with a single parameter of
-         *                 type `Office.AsyncResult`.
+         * @param callback - When the method completes, the function passed in the `callback` parameter is called with a single parameter, `asyncResult`,
+         *                   which is an `Office.AsyncResult` object. The ID of the message is returned in the `asyncResult.value` property.
```
Comment (ElizabethSamuel-MSFT) [inline types/office-js-preview/index.d.ts:16689]:
```suggestion
         *                   which is an `Office.AsyncResult` object. The message ID is returned in the `asyncResult.value` property.
```

File: types/office-js-preview/index.d.ts [hunk 6]
```diff
@@ -16711,12 +16699,29 @@ declare namespace Office {
...
+         * there, `saveAsync` saves the draft to the **Drafts** folder of the shared mailbox.
+         *
          * **Errors**:
          *
          * - `InvalidAttachmentId`: The attachment identifier does not exist.
          *
-         * @param callback - When the method completes, the function passed in the `callback` parameter is called with a single parameter of
-         *                 type `Office.AsyncResult`.
+         * @param callback - When the method completes, the function passed in the `callback` parameter is called with a single parameter, `asyncResult`,
+         *                   which is an `Office.AsyncResult` object. The ID of the message is returned in the `asyncResult.value` property.
```
Comment (ElizabethSamuel-MSFT) [inline types/office-js-preview/index.d.ts:16724]:
```suggestion
         *                   which is an `Office.AsyncResult` object. The message ID is returned in the `asyncResult.value` property.
```

**Review by ElizabethSamuel-MSFT — COMMENTED** [review]
also apply to office-js

**Review by ElizabethSamuel-MSFT — APPROVED** [review]

**Pull request merged**